File: app/src/screens/BlockedUsersScreen.js
// ENGELLENEN KULLANICILAR — kullanıcının kendi engel listesi (E9).
//
//   • Liste sunucudan gelir ve YALNIZ kendi engellerin görünür.
//   • "Beni kim engelledi" diye bir ekran YOKTUR ve olmayacaktır: engel karşı
//     tarafa ilan edilmez. Bunu göstermek, engelin kendisini bir tacize
//     dönüştürürdü.
//   • Engel geri alınabilir. Google Play, engelleme sunan uygulamalarda geri
//     almanın da uygulama içinden erişilebilir olmasını bekler.
//   • Hesabı silinmiş kişi listeden DÜŞMEZ, "Silinmiş kullanıcı" diye
//     etiketlenir. Satırı gizlemek, kullanıcıya engelinin kendiliğinden
//     kalktığı izlenimini verirdi.
import React, { useCallback, useEffect, useState } from 'react';
import { View, Text, TouchableOpacity, StyleSheet, ScrollView, ActivityIndicator, RefreshControl } from 'react-native';
import { colors, spacing, radius } from '../theme';
import ScreenBackdrop from '../components/ScreenBackdrop';
import { CommentAvatar } from '../components';
import { api } from '../api';

function tarih(iso) {
  if (!iso) return '—';
  try { return new Date(iso).toLocaleDateString('tr-TR'); } catch { return '—'; }
}

export default function BlockedUsersScreen() {
  const [data, setData] = useState(null);
  const [refreshing, setRefreshing] = useState(false);
  const [err, setErr] = useState(null);
  const [busyId, setBusyId] = useState(null);

  const load = useCallback(async () => {
    setErr(null);
    try { setData(await api.blocks()); }
    catch (e) { setErr(e.message); }
  }, []);
  useEffect(() => { load(); }, [load]);

  const refresh = async () => { setRefreshing(true); await load(); setRefreshing(false); };

  const kaldir = async (userId) => {
    setBusyId(userId); setErr(null);
    try { await api.unblockUser(userId); await load(); }
    catch (e) { setErr(e.message); } finally { setBusyId(null); }
  };

  const list = data?.blocks || [];

  return (
    <ScreenBackdrop>
      <ScrollView
        style={styles.container}
        contentContainerStyle={{ padding: spacing.lg }}
        refreshControl={<RefreshControl refreshing={refreshing} onRefresh={refresh} tintColor={colors.primary} />}
      >
        <Text style={styles.title}>Engellenen Kullanıcılar</Text>
        <Text style={styles.help}>
          Engellediğin kişilerin yorumlarını görmezsin, onlar da seninkileri göremez.
          Engellediğin kişiye bildirim gitmez. İstediğin zaman engeli kaldırabilirsin.
        </Text>

        {err && <Text style={styles.err}>{err}</Text>}
        {data === null && !err && <ActivityIndicator size="large" color={colors.primary} style={{ marginTop: spacing.lg }} />}

        {list.map((b) => (
          <View key={b.userId} style={styles.card}>
            <CommentAvatar
              size={38}
              author={{ username: b.username, avatarType: b.avatarType, avatarKey: b.avatarKey, avatarUrl: b.avatarUrl }}
            />
            <View style={{ flex: 1 }}>
              <Text style={styles.name} numberOfLines={1}>{b.username}</Text>
              <Text style={styles.meta}>Engellendi: {tarih(b.createdAt)}</Text>
            </View>
            <TouchableOpacity
              style={styles.unblockBtn}
              onPress={() => kaldir(b.userId)}
              disabled={busyId === b.userId}
            >
              {busyId === b.userId
                ? <ActivityIndicator size="small" color={colors.primary} />
                : <Text style={styles.unblockTxt}>Engeli Kaldır</Text>}
            </TouchableOpacity>
          </View>
        ))}

        {data && list.length === 0 && (
          <Text style={styles.note}>
            Engellediğin kimse yok. Bir kullanıcıyı, yorumunun altındaki “Engelle”
            düğmesinden engelleyebilirsin.
          </Text>
        )}
      </ScrollView>
    </ScreenBackdrop>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: 'transparent' },
  title: { color: colors.text, fontSize: 22, fontWeight: '800', marginBottom: 6 },
  help: { color: colors.textMuted, fontSize: 12.5, lineHeight: 18, marginBottom: spacing.md },
  note: { color: colors.textMuted, fontSize: 12.5, lineHeight: 18, textAlign: 'center', marginTop: spacing.md },
  err: { color: colors.red, fontSize: 13, fontWeight: '600', marginBottom: spacing.sm },
  card: { flexDirection: 'row', alignItems: 'center', gap: 12, backgroundColor: colors.card, borderRadius: radius.md, borderWidth: 1, borderColor: colors.border, padding: spacing.md, marginBottom: spacing.sm },
  name: { color: colors.text, fontSize: 14.5, fontWeight: '800' },
  meta: { color: colors.textMuted, fontSize: 11.5, marginTop: 3 },
  unblockBtn: { borderWidth: 1, borderColor: colors.primary, borderRadius: radius.sm, paddingHorizontal: 10, paddingVertical: 7 },
  unblockTxt: { color: colors.primary, fontSize: 12, fontWeight: '800' },
});

File: flutter/lib/features/profile/blocked_users_screen.dart
// KAYNAK: app/src/screens/BlockedUsersScreen.js — BİREBİR çeviri.
//
// ENGELLENEN KULLANICILAR — kullanıcının kendi engel listesi (E9).
//
//   • Liste sunucudan gelir ve YALNIZ kendi engellerin görünür.
//   • "Beni kim engelledi" diye bir ekran YOKTUR ve olmayacaktır: engel karşı
//     tarafa ilan edilmez. Bunu göstermek, engelin kendisini bir tacize
//     dönüştürürdü.
//   • Engel geri alınabilir. Google Play, engelleme sunan uygulamalarda geri
//     almanın da uygulama içinden erişilebilir olmasını bekler.
//   • Hesabı silinmiş kişi listeden DÜŞMEZ, "Silinmiş kullanıcı" diye
//     etiketlenir. Satırı gizlemek, kullanıcıya engelinin kendiliğinden
//     kalktığı izlenimini verirdi.

import 'package:flutter/material.dart';

import '../../core/network/api_client.dart';
import '../../core/theme/tokens.dart';
import '../../widgets/avatar.dart';
import '../../widgets/screen_backdrop.dart';
import 'devices_screen.dart' show trTarih;

class BlockedUsersScreen extends StatefulWidget {
  const BlockedUsersScreen({super.key});

  @override
  State<BlockedUsersScreen> createState() => _BlockedUsersScreenState();
}

class _BlockedUsersScreenState extends State<BlockedUsersScreen> {
  Map<String, dynamic>? _data;
  String? _err;
  Object? _busyId;

  @override
  void initState() {
    super.initState();
    _load();
  }

  Future<void> _load() async {
    if (mounted) setState(() => _err = null);
    try {
      final d = await api.blocks();
      if (mounted) setState(() => _data = (d as Map).cast<String, dynamic>());
    } catch (e) {
      if (mounted) setState(() => _err = '$e');
    }
  }

  Future<void> _kaldir(Object userId) async {
    setState(() {
      _busyId = userId;
      _err = null;
    });
    try {
      await api.unblockUser(userId);
      await _load();
    } catch (e) {
      if (mounted) setState(() => _err = '$e');
    } finally {
      if (mounted) setState(() => _busyId = null);
    }
  }

  @override
  Widget build(BuildContext context) {
    final list = (_data?['blocks'] as List?) ?? const [];

    return Scaffold(
      appBar: AppBar(title: const Text('Engellenen Kullanıcılar')),
      body: ScreenBackdrop(
        child: RefreshIndicator(
          color: AppColors.primary,
          onRefresh: _load,
          child: ListView(
            padding: const EdgeInsets.all(Spacing.lg),
            children: [
              Text(
                'Engellenen Kullanıcılar',
                style: TextStyle(
                  color: AppColors.onBackground,
                  fontSize: 22,
                  fontWeight: AppFont.heavy,
                ),
              ),
              const SizedBox(height: 6),
              Text(
                'Engellediğin kişilerin yorumlarını görmezsin, onlar da seninkileri göremez. '
                'Engellediğin kişiye bildirim gitmez. İstediğin zaman engeli kaldırabilirsin.',
                style: TextStyle(
                  color: AppColors.textMuted,
                  fontSize: 12.5,
                  height: 18 / 12.5,
                ),
              ),
              const SizedBox(height: Spacing.md),
              if (_err != null)
                Padding(
                  padding: const EdgeInsets.only(bottom: Spacing.sm),
                  child: Text(
                    _err!,
                    style: const TextStyle(
                      color: AppColors.red,
                      fontSize: 13,
                      fontWeight: AppFont.semibold,
                    ),
                  ),
                ),
              if (_data == null && _err == null)
                Padding(
                  padding: EdgeInsets.only(top: Spacing.lg),
                  child: Center(
                    child: CircularProgressIndicator(color: AppColors.primary),
                  ),
                ),
              for (final b in list) _satir((b as Map).cast<String, dynamic>()),
              if (_data != null && list.isEmpty)
                Padding(
                  padding: EdgeInsets.only(top: Spacing.md),
                  child: Text(
                    'Engellediğin kimse yok. Bir kullanıcıyı, yorumunun altındaki “Engelle” '
                    'düğmesinden engelleyebilirsin.',
                    textAlign: TextAlign.center,
                    style: TextStyle(
                      color: AppColors.textMuted,
                      fontSize: 12.5,
                      height: 18 / 12.5,
                    ),
                  ),
                ),
            ],
          ),
        ),
      ),
    );
  }

  Widget _satir(Map<String, dynamic> b) {
    final id = b['userId'];
    return Container(
      margin: const EdgeInsets.only(bottom: Spacing.sm),
      padding: const EdgeInsets.all(Spacing.md),
      decoration: BoxDecoration(
        color: AppColors.card,
        borderRadius: BorderRadius.circular(AppRadius.md),
        border: Border.all(color: AppColors.border),
      ),
      child: Row(
        children: [
          CommentAvatar(
            size: 38,
            author: {
              'username': b['username'],
              'avatarType': b['avatarType'],
              'avatarKey': b['avatarKey'],
              'avatarUrl': b['avatarUrl'],
            },
          ),
          const SizedBox(width: 12),
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  '${b['username']}',
                  maxLines: 1,
                  overflow: TextOverflow.ellipsis,
                  style: TextStyle(
                    color: AppColors.text,
                    fontSize: 14.5,
                    fontWeight: AppFont.heavy,
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.only(top: 3),
                  child: Text(
                    'Engellendi: ${trTarih(b['createdAt'])}',
                    style: TextStyle(
                      color: AppColors.textMuted,
                      fontSize: 11.5,
                    ),
                  ),
                ),
              ],
            ),
          ),
          const SizedBox(width: 12),
          GestureDetector(
            onTap: _busyId == id ? null : () => _kaldir(id as Object),
            child: Container(
              padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 7),
              decoration: BoxDecoration(
                borderRadius: BorderRadius.circular(AppRadius.sm),
                border: Border.all(color: AppColors.primary),
              ),
              child: _busyId == id
                  ? SizedBox(
                      width: 14,
                      height: 14,
                      child: CircularProgressIndicator(
                        strokeWidth: 2,
                        color: AppColors.primary,
                      ),
                    )
                  : Text(
                      'Engeli Kaldır',
                      style: TextStyle(
                        color: AppColors.primary,
                        fontSize: 12,
                        fontWeight: AppFont.heavy,
                      ),
                    ),
            ),
          ),
        ],
      ),
    );
  }
}
